src/Trie.mo

```diff
@@ -87,35 +92,16 @@ public type Key<K> = {
   key: K;
 };
 
+public type List<T> = List.List<T>;
```

Comment:
Why is `Trie` reexporting the `List` type? (Should this maybe be `private`?)

Reply:
Perhaps?  I will do so here, since I tend to agree.

(OTOH, I hesitate slightly because the `ListUtils` submodule uses this type in its public API.  If we just import `Trie` and want to refer to these types, it's arguably useful for making the API more self-contained?  Perhaps that's why it was like this, not sure.)